File: DSA/AnshulJain_590014934/Day_99_Question_3.py
class Node:
    def __init__(self, val):
        self.val = val
        self.left = None
        self.right = None

def build_tree(values):
    nodes = [None if v == "null" else Node(int(v)) for v in values]
    kids = nodes[::-1]
    root = kids.pop()
    for node in nodes:
        if node:
            if kids: node.left = kids.pop()
            if kids: node.right = kids.pop()
    return root

def max_path_sum(root):
    best = float("-inf")
    def dfs(node):
        nonlocal best
        if not node: return 0
        left = max(dfs(node.left), 0)
        right = max(dfs(node.right), 0)
        best = max(best, node.val + left + right)
        return node.val + max(left, right)
    dfs(root)
    return best

values = input("Enter tree nodes in level order (use 'null' for empty): ").split()
root = build_tree(values)
print(max_path_sum(root))

File: DSA/GEHNA_KHANNA_590013140/day99question3.c
#include <stdio.h>
#include <stdlib.h>
#include <limits.h>

struct Node {
    int val;
    struct Node *left, *right;
};

struct Node* newNode(int val) {
    struct Node* node = (struct Node*)malloc(sizeof(struct Node));
    node->val = val;
    node->left = node->right = NULL;
    return node;
}

int max(int a, int b) { return a > b ? a : b; }

int maxPathUtil(struct Node* root, int *res) {
    if (!root) return 0;
    int l = maxPathUtil(root->left, res);
    int r = maxPathUtil(root->right, res);
    int max_single = max(max(l, r) + root->val, root->val);
    int max_top = max(max_single, l + r + root->val);
    if (*res < max_top) *res = max_top;
    return max_single;
}

int maxPathSum(struct Node* root) {
    int res = INT_MIN;
    maxPathUtil(root, &res);
    return res;
}

struct Node* buildTree(int n) {
    struct Node* nodes[n];
    for (int i = 0; i < n; i++) {
        int val; scanf("%d", &val);
        if (val == -1) nodes[i] = NULL;
        else nodes[i] = newNode(val);
    }
    for (int i = 0, j = 1; j < n; i++) {
        if (nodes[i]) {
            if (j < n) nodes[i]->left = nodes[j++];
            if (j < n) nodes[i]->right = nodes[j++];
        }
    }
    return nodes[0];
}

int main() {
    int n;
    scanf("%d", &n);
    struct Node* root = buildTree(n);
    printf("%d\n", maxPathSum(root));
    return 0;
}
